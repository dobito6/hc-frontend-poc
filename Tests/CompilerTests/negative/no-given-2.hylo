trait P {}

struct A { memberwise init }
struct B { memberwise init }

given A: P {}

fun f<T where T: P, A: P>(x: T) {}

fun main() {
  f(x: B())
}

