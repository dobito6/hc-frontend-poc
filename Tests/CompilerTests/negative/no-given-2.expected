negative/no-given-2.hylo:11.3-4: error: no coercion from '<T> (P<T>, P<A>) ==> [Void](x: let T) let -> Void' to '[Void](x: let B) let -> Void'
  f(x: B())
  ^
negative/no-given-2.hylo:11.3-4: note: no given instance of 'P<B>' in this scope
  f(x: B())
  ^
